### PR TITLE
Multi Domain First Parties

### LINKED ISSUE #781: Add list of multi-domain first parties
At the Wikimedia Foundation, we run web sites on a number of different domains that talk to each other and are essentially all the same first party. Privacy Badger recognizes that `en.wikipedia.org` and `de.wikipedia.org` are the same party, but it does not recognize that domains like `commons.wikimedia.org` and `fr.wiktionary.org` are also the same party. This frequently leads Privacy Badger to block AJAX requests from e.g. `*.wikipedia.org` domains to `*.wiktionary.org` domains, which breaks [our unified login system](https://phabricator.wikimedia.org/T108505) and [our new cross-wiki notifications beta feature](https://phabricator.wikimedia.org/T129764).

Putting our domains on the yellow list doesn't help, because these features are closely tied to user identity so they need cookies for authentication to work. Another way we could work around this is by publishing a DNT policy file, but we can't, at least not right now, since we have our own privacy policy that is almost-but-not-quite DNT-compliant (see [relevant policy paragraph](https://wikimediafoundation.org/wiki/Privacy_policy#Our_Response_to_Do_Not_Track_.28DNT.29_signals) and [FAQ  entry](https://wikimediafoundation.org/wiki/Privacy_policy/FAQ#DNTFAQ)).

I've been thinking that other web sites probably have a similar issue. For example, a request from `gmail.com` to `*.google.com` should also be treated as a first-party request rather than a third-party request. For that reason I propose adding a list of groups of domains that all belong to the same party, and have Privacy Badger treat a cross-domain request within the same domain group as a first-party request (i.e. allow it by default).

### PR DESCRIPTION
**Pull request by cooperq**
Fixes #781 adds awareness of domains which are techincally third party but actually belong to the same first party. 

cc @cowlicks @Hainish @alexristich

### CODE REVIEW

File: src/multiDomainFirstParties.js [hunk 1]
```diff
@@ -11,15 +11,50 @@
...
+  ["wikia.com", "wikia.net", "nocookie.net"],
+  ["tripadvisor.com", "tacdn.com", "tamgrt.com"],
+  ["baidu.com", "bdimg.com", "bdstatic.com"],
+  ["wikipedia.org", "wikimedia.org", "wikimediafoundation.org", "wiktionary.org", 
+    "wikiquote.org", "wikibooks.org", "wikisource.org", "wikinews.org", 
+    "wikiversity.org", "mediawiki.org", "wikidata.org", "wikivoyage.org"],
+  ["twitter.com", "twimg.com", "t.co"],
+  [ "amazon.com", "amazon.ca", "amazon.co.jp", "amazon.co.uk", "amazon.de", 
+    "amazon.es", "amazon.fr", "amazon.it"],
+  ["pornhub.com", "phncdn.com"],
```
Comment (cowlicks) [inline src/multiDomainFirstParties.js]: https://www.youtube.com/watch?v=3gOHvDP_vCs

Comment (cooperq) [inline src/multiDomainFirstParties.js]: Just giving the people what they want ;)

**Comment by alexristich** [comment]
Took a look at the code and it looks good. Is "dummy" there as a placeholder for additional properties to be added, or was that primarily for testing? 

For the array itself, is it better to have this within the code for the extension or host it on the EFF site in a JSON file like the DNT-compliance list and refresh it daily?

**Comment by cooperq** [comment]
Dummy is there just so we don't have to remember to add or remove a comma at the end each time. IDK, it could be removed if it's too confusing. 
We could have the first parties array in a JSON file, it really depends how often we think we are going to have to update it. I suspect it won't have to get updated that often, but if that turns out to not be the case I think it should be pretty easy to refactor.

**Comment by alexristich** [comment]
True, true. I suspect it might occur somewhat frequently judging by some of the recent tickets, but as you say, refactoring shouldn't be much of an issue down the line.

**Comment by cowlicks** [comment]
LGTM. It'd be nice to have an integration test on pbtest.org.

**Pull request merged**